functions/tests/test_stripe_webhooks.py

```
@@ -0,0 +1,560 @@
+"""
+Comprehensive end-to-end tests for Stripe webhook functionality.
+Tests the complete webhook handling workflow including session processing,
+fulfillment, and error handling scenarios.
+"""
+
+import hashlib
```

Ruff (0.12.2)

`hashlib` imported but unused

Remove unused import: `hashlib`

(F401)